src/component/feature/FeatureStrategy/FeatureStrategyForm/FeatureStrategyForm.tsx

```diff
@@ -0,0 +1,158 @@
+        event.preventDefault();
+        if (enableProdGuard) {
+            setShowProdGuard(true);
+        } else {
+            onSubmit();
+        }
+    };
+
+    const hasValidConstraints = useConstraintsValidation(
+        feature.project,
+        feature.name,
+        strategy.constraints
+    );
+
+    // TODO(olav): Remove uiConfig.flags.CO when new constraints are released.
+    const FeatureStrategyConstraintsImplementation = uiConfig.flags.CO
+        ? FeatureStrategyConstraints2
+        : FeatureStrategyConstraints;
+    const disableSubmitButtonFromConstraints = uiConfig.flags.CO
+        ? !hasValidConstraints
+        : false;
+
+    return (
+        <form className={styles.form} onSubmit={onSubmitOrProdGuard}>
+            <h2 className={styles.title}>
+                <StrategyIcon className={styles.icon} aria-hidden />
+                <span className={styles.name}>{strategyName}</span>
+            </h2>
+            <div>
+                <FeatureStrategyEnabled
+                    feature={feature}
+                    environmentId={environmentId}
+                />
+            </div>
+            <ConditionallyRender
+                condition={Boolean(uiConfig.flags[C])}
+                show={
+                    <div>
+                        <FeatureStrategyConstraintsImplementation
+                            projectId={feature.project}
+                            environmentId={environmentId}
+                            strategy={strategy}
+                            setStrategy={setStrategy}
+                            mode={mode}
+                        />
+                    </div>
+                }
+            />
+            <FeatureStrategyType
+                strategy={strategy}
+                setStrategy={setStrategy}
+                hasAccess={hasAccess}
+            />
+            <div className={styles.buttons}>
+                <Button
+                    type="button"
+                    color="secondary"
+                    onClick={onCancel}
+                    disabled={loading}
+                >
+                    Cancel
+                </Button>
+                <ConditionallyRender
+                    condition={Boolean(strategy.id)}
+                    show={
+                        <FeatureStrategyRemove
+                            projectId={feature.project}
+                            featureId={feature.name}
+                            environmentId={environmentId}
+                            strategyId={strategy.id!}
+                            disabled={loading}
+                        />
+                    }
+                />
+                <FeatureStrategyProdGuard
+                    open={showProdGuard}
+                    onClose={() => setShowProdGuard(false)}
+                    onClick={onSubmit}
+                    loading={loading}
+                    label="Save strategy"
+                />
+                <Button
```

Comment:
We might want to have this as a PermissionButton (in case someone is linking directly to this page).